concepts/pipeline/http_edge_spec.md

```diff
@@ -1,16 +1,75 @@
 
-## GET Requests
-Accept GET on `/status`, returning `OK` if all is well. This can be used to check the health of web servers.
+HTTP submissions come in from the wild, hit a load balancer, then optionally an
+Nginx proxy, then the HTTP Edge Server described in this document. Data is
+accepted via a POST/PUT request from clients, which the server will wrap in a
+[Heka message](http://mozilla-services.github.io/lua_sandbox/heka/message.html)
+and forward to two places: the [Services Data Pipeline](data_pipeline.md), where
+any further processing, analysis, and storage will be handled; as well as to a
+short-lived S3 bucket which will act as a fail-safe in case there is a
+processing error and/or data loss within the main Data Pipeline.
 
-## POST/PUT Requests
+### Namespaces
+
+Namespaces are used to control the processing of data from different types of
+clients, from the metadata that is collected to the destinations where the data
+is written, processed and accessible. Namespaces are
+[configured](https://github.com/mozilla-services/nginx_moz_ingest#example-configuration)
+in Nginx using a location directive, to request a new namespace file a bug
+against the [Data Platform Team](https://bugzilla.mozilla.org/enter_bug.cgi?product=Data%20Platform%20and%20Tools&component=Pipeline%20Ingestion)
+with a short description of what the namespace will be used for and the desired
+configuration options.
+
+### Forwarding to the pipeline
+
+The constructed Heka protobuf message to is written to disk and the pub/sub
+pipeline (currently Kafka). The messages written to disk serve as a fail-safe,
+they are batched and written to S3 (landfill) when they reach a certain size or
+timeout.
+
+#### Edge Server Heka Message Schema
+
+* required binary **Uuid**;                     // Internal identifier randomly generated
+* required int64  **Timestamp**;                // Submission time (server clock)
+* required string **Hostname**;                 // Hostname of the edge server e.g. `ip-172-31-2-68`
+* required string **Type**;                     // Kafka topic name e.g. `telemetry-raw`
+* required group  **Fields**
+   * required string **uri**;                   // Submission URI e.g. `/submit/telemetry/6c49ec73-4350-45a0-9c8a-6c8f5aded0cf/main/Firefox/58.0.2/release/20180206200532`
+   * required binary **content**;               // POST Body
+   * required string **protocol**;              // e.g. `HTTP/1.1`
+   * optional string **args**;                  // Query parameters e.g. `v=4`
+   * optional string **remote_addr**;           // In our setup it is usually a load balancer e.g. `172.31.32.229`
```

Comment:
Per the [contributing guidelines](https://docs.telemetry.mozilla.org/meta/contributing.html#spell-checking), please enclose field names with backticks rather than adding them to the `.spelling` file.